scene/gui/texture_rect.cpp

```diff
@@ -40,23 +40,37 @@ void TextureRect::_notification(int p_what) {
 
 		switch (stretch_mode) {
 			case STRETCH_SCALE_ON_EXPAND: {
-				Size2 s = expand ? get_size() : texture->get_size();
-				draw_texture_rect(texture, Rect2(Point2(), s), false);
+				Size2 size = expand ? get_size() : texture->get_size();
+				size.x *= hflip ? -1.0f : 1.0f;
+				size.y *= vflip ? -1.0f : 1.0f;
+				draw_texture_rect(texture, Rect2(Point2(), size), false);
 			} break;
 			case STRETCH_SCALE: {
-				draw_texture_rect(texture, Rect2(Point2(), get_size()), false);
+				Size2 size = get_size();
+				size.x *= hflip ? -1.0f : 1.0f;
+				size.y *= vflip ? -1.0f : 1.0f;
+				draw_texture_rect(texture, Rect2(Point2(), size), false);
```

Comment:
Seems like each `case` is doing the same work in this `switch`. Can we do anything to prevent repeating ourselves?

Reply:
Do you mean for the sake of having the flipping logic in just one place?
I can replace these two lines:
```
size.x *= hflip ? -1.0f : 1.0f;
size.y *= vflip ? -1.0f : 1.0f;
```
with a function like:
```
flip_size(size);
```
```
void flip_size(Size2 & size) {
    size.x *= hflip ? -1.0f : 1.0f;
    size.y *= vflip ? -1.0f : 1.0f;
}
```
Would that help?

Reply:
You can probably get the `draw_texture_rect` call out of the `switch` statement, and have the flip code right before it, no?

Reply:
Yep, I can do that instead. Although, because the last case calls `draw_texture_rect_region` instead of `draw_texture_rect` I'll have to add another condition outside of the switch to handle it. Is it ok?

Reply:
Yeah, either that, or just have the other parameters of `draw_texture_rect_region` be variables as well.